pwndbg/emu/emulator.py

```diff
@@ -220,12 +220,28 @@ def __init__(self) -> None:
         # (address_successfully_executed, size_of_instruction)
         self.last_single_step_result = InstructionExecutedResult(None, None)
 
-        # Initialize the register state
-        for reg in (
-            list(self.regs.retaddr)
+        # The specific order of this list is very important:
+        # Due to the behavior of Arm in the Unicorn engine,
+        # we must write the flags register after PC, and the stack pointer after the flags register.
+        # Otherwise, the values will be clobbered
+        # https://github.com/pwndbg/pwndbg/pull/2337
+        raw_list = (
+            [self.regs.pc]
+            + list(self.regs.flags)
+            + [self.regs.stack, self.regs.frame]
+            + list(self.regs.retaddr)
             + list(self.regs.misc)
-            + list(self.regs.common)  # this includes the flags register
-        ):
+            + list(self.regs.gpr)
+        )
+
+        # Get rid of duplicates and "None" values - some registers, like .frame are sometimes non-existent
+        reg_list = []
+        for x in raw_list:
+            if x and x not in reg_list:
+                reg_list.append(x)
+
+        # Initialize the register state
```

Comment:
These lines mimic the register initialization loop in pwndbg/lib/regs.py.

We are computing this every time the emulator is instantiated - should we instead make a variable like `emulated_registers` that are in this order in the RegisterSet object?

Reply:
Haha, commented on that. Generally, yes. Pls do this in another PR :)

Reply:
(or in this PR if it isn't merged yet)

Reply:
Great! Last commits do this